src/Neo.IO/ByteArrayComparer.cs

```diff
@@ -29,7 +29,6 @@ private ByteArrayComparer(int direction)
 
         public int Compare(byte[]? x, byte[]? y)
         {
-            if (ReferenceEquals(x, y)) return 0;
```

Comment:
But now x==y will iterate all the entries

Reply:
Can you fix it? But dont use https://github.com/neo-project/neo/pull/3136#discussion_r1487481655 because the `internal` method doesn't allow `null` we will get exceptions.

Reply:
maybe do `x is not null when y not null && x.Equal(y)`

Reply:
I made one last attempt.